kernel/core/maths/matfuncs.hpp

```diff
@@ -3567,6 +3567,174 @@ static Matrix matrixFilter(const MatFuncData& funcData, const MatFuncErrorInfo&
+    else
+        return circShiftCol(funcData, errorInfo);
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Function to perform a shift
+/// of the columns in a matrix by n columns.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix vectShiftCol(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Matrix data is a column-wise vector
+    std::vector<mu::value_type> matData = funcData.mat1.data();
+
+    // Column shift to the right (negative values to the left)
+    for (size_t col = 0; col < funcData.mat1.cols(); col++)
+    {
+        // TODO: Remove the unnecessary int casts
+        int col_old = (int)((((int)col - (int)funcData.nVal) % (int)funcData.mat1.cols()) + (int)funcData.mat1.cols()) % (int)funcData.mat1.cols();
+        std::copy(funcData.mat1.data().begin() + col_old * funcData.mat1.rows(), funcData.mat1.data().begin() + (col_old + 1) * funcData.mat1.rows(), matData.begin() + col * funcData.mat1.rows());
+    }
+
+    Matrix _mResult = Matrix();
+    _mResult.assign(funcData.mat1.rows(), funcData.mat1.cols(), matData);
+
+    return _mResult;
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Function to perform a shift
+/// of the rows in a matrix by n rows.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix vectShiftRow(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Transpose the matrix (including restructuring the memory) to get a row wise representation of the data
+    Matrix _mResult = Matrix(funcData.mat1);
+    _mResult.transpose();
+    _mResult.extend(_mResult.rows(), _mResult.cols());
+
+    MatFuncData funcDataTransposed = MatFuncData(_mResult, funcData.nVal);
+
+    // Apply the vectShiftRow to the transposed matrix
+    _mResult = vectShiftCol(funcDataTransposed, errorInfo);
+
+    // Reverse the transposition
+    _mResult.transpose();
+    _mResult.extend(_mResult.rows(), _mResult.cols());
+
+    // TODO: The double transpose is not very efficient. Probably a loop with directly addressing the new elements would be faster.
+
+    return _mResult;
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Wrapper for the two vector shift
+/// functions. Selects the methode and checks
+/// the dimension argument.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix vectShift(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Check the dim argument
+    if (funcData.mVal < 0 || funcData.mVal > 1)
+        throw SyntaxError(SyntaxError::INVALID_AXIS, errorInfo.command, errorInfo.position);

```

Comment:
Whether a check like this is actually necessary, is debatable. Especially considering, that the other parameter count is checked nowhere.

Reply:
I do not know if it is really necessary, but could be helpful and also I would not know how to handle different inputs here. For the other parameter there are no invalid inputs, so no handling is required. 

Okay, complex numbers would be invalid input for the numbers of positions to shift, I forgot about that.